src/main/java/com/laborete/LaboreteAPI/services/UsersServiceImpl.java

```diff
@@ -50,4 +63,62 @@ public UserEntity createUser(UserEntity user) {
         return this.usersRepository.save(user);
     }
 
+    public String uploadUserAvatar(String name, MultipartFile file, UUID userId) {
+        UserEntity user = this.usersRepository.getUserById(userId);
+
+        if (file.isEmpty()) {
+            throw new ResourceBadRequestException(FILE_IS_EMPTY);
+        }
+        if (name.isBlank() || name.isEmpty()) {
+            throw new ResourceBadRequestException(FILE_NAME_IS_EMPTY);
+        }
+        if (user == null) {
+            throw new ResourceNotFoundException(USER_NOT_FOUND + userId);
+        }
+
+        try {
+            byte[] bytes = file.getBytes();
+
+            UUID uuid = UUID.randomUUID();
+            String path = this.createDirectory(uuid);
+
+            File avatar = new File(path, uuid + ".png");
+            BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(avatar));
+
+            stream.write(bytes);
+            stream.close();
+
+            UserAvatarEntity userAvatarEntity = new UserAvatarEntity(uuid, name, file.getSize());
+            user.setUserAvatarEntity(userAvatarEntity);
+            this.usersRepository.save(user);
```

Comment:
The following case is possible: user is successfully saved and during avatar saving an error occured. So, we should make this code transactional. Please use `@Transactional` annotation.